### PR TITLE
Closes AgileVentures/MetPlus_tracker#67

### PR DESCRIPTION
**Pull request by vijayakarumudi**
PR for job_categories: AgileVentures/Metplus#issue67

### CODE REVIEW

**Comment by joaopapereira** [comment]
Connected to AgileVentures/MetPlus_tracker#67

**Comment by joaopapereira** [comment]
Closes AgileVentures/MetPlus_tracker#67

File: app/models/job_category.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+class JobCategory < ActiveRecord::Base
+ has_many :job_specialities
```
Comment (joaopapereira) [inline app/models/job_category.rb]: Shouldn't this be
Has_many :agency_people, through :job_specialities

File: app/models/job_category.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+ has_many :agency_poeple, through: :job_specialities
```
Comment (joaopapereira) [inline app/models/job_category.rb]: Incorrect name agency_people

**Comment by joaopapereira** [comment]
Please review my comments. This issue is connected AgileVentures/MetPlus_tracker#81

File: app/models/job_category.rb (hunk 2 above)
Comment (patmbolger) [inline app/models/job_category.rb]: We have confirmed that 'job_specialities' is NOT a join table but instead is a simple join table.  Therefore this association should read: 

has_and_belongs_to_many :agency_people,  join_table: :job_specialities

File: db/migrate/20151027162612_create_job_categories.rb [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+      t.references :jobs, index: true, foreign_key: true
+      t.references :skills, index: true, foreign_key: true
+
+      t.references :job_speciliaties, index: true, foreign_key: true
+
+      t.timestamps null: false
+    end
+  end
+end
+
```
Comment (patmbolger) [inline db/migrate/20151027162612_create_job_categories.rb]: The "t.references" are incorrect.  Each of these will put a foreign key in the table, to reflect "belongs_to" associations.  

There is no "belongs_to" association for the JobCategory model.

File: app/models/job_category.rb (hunk 1 above)
Comment (patmbolger) [inline app/models/job_category.rb]: This line should be removed - see note below re job_specialities.

File: spec/models/job_category_spec.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
+require 'rails_helper'
+
+RSpec.describe JobCategory, type: :model do
+
+   
+   describe 'Associations' do
+
+     
+     xit { is_expected.to have_many :job_specialities }
```
Comment (patmbolger) [inline spec/models/job_category_spec.rb]: This line should be removed to reflect changed model association (see comment above)

File: spec/models/job_category_spec.rb [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+     xit { is_expected.to have_many(:agency_people).through(:job_specialities)}
```
Comment (patmbolger) [inline spec/models/job_category_spec.rb]: Please change to:

it { is_expected.to have_many(:agency_people).join_table(:job_specialities)}

File: spec/models/job_category_spec.rb [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+     xit { is_expected.to has_and_belongs_to_many(:skills)}
```
Comment (patmbolger) [inline spec/models/job_category_spec.rb]: the matcher is 'have_many_and_belongs_to'  (this line has 'has_ ....')

**Comment by vijayakarumudi** [comment]
close this PR

**Pull request closed without merging**